test/integration/framework/master_utils.go

```diff
@@ -148,28 +150,30 @@ func startMasterOrDie(masterConfig *master.Config) (*master.Master, *httptest.Se
 
 // Returns a basic master config.
 func NewMasterConfig() *master.Config {
-	etcdClient := NewEtcdClient()
-	storageVersions := make(map[string]string)
-
-	etcdStorage := etcdstorage.NewEtcdStorage(etcdClient, testapi.Default.Codec(), etcdtest.PathPrefix(), false)
-	storageVersions[api.GroupName] = testapi.Default.GroupVersion().String()
-	autoscalingEtcdStorage := NewAutoscalingEtcdStorage(etcdClient)
-	storageVersions[autoscaling.GroupName] = testapi.Autoscaling.GroupVersion().String()
-	batchEtcdStorage := NewBatchEtcdStorage(etcdClient)
-	storageVersions[batch.GroupName] = testapi.Batch.GroupVersion().String()
-	expEtcdStorage := NewExtensionsEtcdStorage(etcdClient)
-	storageVersions[extensions.GroupName] = testapi.Extensions.GroupVersion().String()
-
-	storageDestinations := genericapiserver.NewStorageDestinations()
-	storageDestinations.AddAPIGroup(api.GroupName, etcdStorage)
-	storageDestinations.AddAPIGroup(autoscaling.GroupName, autoscalingEtcdStorage)
-	storageDestinations.AddAPIGroup(batch.GroupName, batchEtcdStorage)
-	storageDestinations.AddAPIGroup(extensions.GroupName, expEtcdStorage)
+	etcdConfig := etcdstorage.EtcdConfig{
+		ServerList: []string{"http://127.0.0.1:4001"},
+		Prefix:     etcdtest.PathPrefix(),
+	}
+
+	negotiatedSerializer := serializer.NewNegotiatedSerializerWapper(api.Scheme, testapi.Default.Codec(), "application/json")
```

Comment:
> I'm not really following-- I don't see any tests that use this?

All the integration tests via `NewMasterConfig` https://github.com/kubernetes/kubernetes/blob/master/test/integration/framework/master_utils.go#L150